statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,534 @@
+
+        if offset is None:
+            offset = getattr(self, 'offset', 0)
+
+        fitted = np.dot(exog, params[:exog.shape[1]])
+        linpred = fitted + exposure + offset
+
+        if which == 'mean':
+            return np.exp(linpred) / (1 - np.exp(np.exp(-linpred)))
+        elif which == 'linear':
+            return linpred
+        elif which =='prob':
+            counts = np.atleast_2d(np.arange(0, np.max(self.endog)+1))
+            mu = self.predict(params, exog=exog, exposure=exposure,
+                              offset=offset)[:,None]
+            return self.model_dist.pmf(counts, mu, self.trunc)
+        else:
+            raise TypeError(
+                    "argument wich == %s not handled" % which)
+
+class TruncatedPoisson(GenericTruncated):
+    """
+    Poisson Zero Inflated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    exog_infl: array
+        A reference to the zero-inflated exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, offset=None, exposure=None,
+                 truncation=0, missing='none', **kwargs):
+        super(TruncatedPoisson, self).__init__(endog, exog, offset=offset,
+                                               exposure=exposure,
+                                               truncation=truncation,
+                                               missing=missing, **kwargs)
+        self.model_main_name = Poisson
+        self.model_main = Poisson(self.endog, self.exog)
+        self.model_dist = truncatedpoisson
+        self.result = TruncatedPoissonResults
+        self.result_wrapper = TruncatedPoissonResultsWrapper
+        self.result_reg = L1TruncatedPoissonResults
+        self.result_reg_wrapper = L1TruncatedPoissonResultsWrapper
+
+class GenericHurdle(CountModel):
+    __doc__ = """
+    Generic Hurdle model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, offset=None,
```

Comment:
I think we should have two exog as in zero-inflated, one for the zeros and one for y > 0